chat-plugins/scavengers.js

```diff
@@ -1,303 +1,612 @@
 /**
-* Scavenger hunts plugin. Only works in a room with the id 'scavengers'.
-* This game shows a hint. Once a player submits the correct answer, they move on to the next hint.
-* You finish upon correctly answering the third hint.
-* In an official hunt, the first three to finish within 60 seconds achieve blitz.
-*/
+ * Scavengers Plugin
+ * Pokemon Showdown - http://pokemonshowdown.com/
+ *
+ * This is a game plugin to host scavenger games specifically in the Scavengers room,
+ * where the players will race answer several hints.
+ *
+ * @license MIT license
+ */
 
 'use strict';
+
 const fs = require('fs');
-let scavengers = {
-	status: 'off',
-	blitz: null,
-	numblitzes: 0,
-	hints: null,
-	answers: null,
-	participants: {},
-	finished: [],
-	resultHTML: null,
-	updatinglb: false,
-};
 
-let scavengersRoom = Rooms.get('scavengers');
-if (scavengersRoom) {
-	if (scavengersRoom.plugin) {
-		scavengers = scavengersRoom.plugin;
-	} else {
-		scavengersRoom.plugin = scavengers;
-	}
-}
+const dataFile = './config/chat-plugins/scavdata.json';
+const points = [20, 15, 10, 5, 1];
+
+let leaderboard = {};
+
+fs.readFile(dataFile, 'utf8', (err, content) => {
+	if (err) return console.log(`ERROR: Unable to load scavenger leaderboard: ${err}`);
 
-let scavengersLB = {};
-let dataFile = './config/chat-plugins/scavdata.json';
-fs.readFile(dataFile, 'utf8', function (err, content) {
-	if (err) {
-		scavengersLB = {};
-		return;
-	}
 	try {
-		scavengersLB = JSON.parse(content);
-	} catch (e) {
-		scavengersLB = {};
-	}
+		leaderboard = JSON.parse(content);
+	} catch (e) {}
 });
-function addPoints(username, points) {
-	let id = toId(username);
-	if (!id) return;
-	if (id in scavengersLB) {
-		scavengersLB[id].points += points;
-	} else {
-		scavengersLB[id] = {
-			points: points,
-			name: Chat.escapeHTML(username),
-		};
-	}
+
+function addPoints(name, points, noUpdate) {
```

Comment:
Hm... maybe static function would be best.  Do you want the ladder function to be a static function in the scavhunt class as well?